exercise12_3.py

```diff
@@ -7,21 +7,18 @@
 Python for Everybody: Exploring Data Using Python 3
 by Charles R. Severance
 
-Solution by Jamison Lahman, June 4, 2017
+Solution by Daniel Lee, Dec 29, 2019
 """
 import urllib.request
-import urllib.parse
-import urllib.error
-
-
-fhand = urllib.request.urlopen('http://data.pr4e.org/romeo.txt')
-
-characters = 0
+url = input('Enter URL: ')
+fhand = urllib.request.urlopen(url)
+count = 0
 for line in fhand:
-    # \n is considered a character
-    # Amend to line.decode().rstrip() if needed
-    words = line.decode()
-    characters = characters + len(words)
-    if characters < 3000:
-        print(line.decode().strip())
-print(characters)
+    sline = line.decode().strip()
+    if count + len(sline) <= 3000:
+        print(sline)
+    elif count < 3000 and count + len(sline) >3000:
+        numchar = 3000 - count
+        print(sline[:numchar - 1])
+    count += len(sline)
+print(count)
```

Comment:
```suggestion
    next_count = count + len(sline)
    if next_count <= 3000:
        print(sline)
    elif count < 3000:
        char_remain = 3000 - count
        print(sline[:char_remain])
    count = next_count
print(count)
```
Otherwise you're doing the same `count += len(sline)` operation 3 times and in 2 different ways. Also, I don't think we want the `-1` otherwise `len(sline[:char_remain - 1]) + count == 3000 - 1`? Additionally, if we're going to use `3000` multiple times, would be better to define it as a constant somewhere (`max_char`).